servers/audio/effects/eq.cpp

```diff
@@ -103,7 +103,12 @@ void EQ::recalculate_band_coefficients() {
 
 		//printf("band %i, precoefs = %f,%f,%f\n",i,c2a,c2b,c2c);
 
-		double r1, r2; //roots
+		// We're assigning r1 an arbitrary value to to avoid the compiler "'r1' may be used uninitialized" warning below.
+		// We're certain it's either initialised in solve_quadratic or solve_quadratic returns 0.
+		// We can't check that the aribtrary value has been changed by solve_quadratic afterwards, because
```

Comment:
Typo in arbitrary.
5 lines of comment for a simple initialization of new variables is a bit overkill IMO. I'd just go with:
```
// Default initializing to silence compiler warning about potential uninitialized use.
// Both variables are properly set in _solve_quadratic before use, or we continue roots == 0.
```

Reply:
I've added an if to: "...or we continue **if** roots == 0."